rust/core-lib/src/editor.rs

```diff
@@ -475,9 +475,24 @@ impl Editor {
             let start = if !region.is_caret() {
                 region.min()
             } else {
-                // TODO: implement complex emoji logic
-                self.text.prev_codepoint_offset(region.end).unwrap_or(region.end)
+                // backspace deletes max(1, tab_size) contiguous spaces
+                let (_, c) = self.view.offset_to_line_col(&self.text,
+                                                          region.start);
+                let use_spaces = self.config.items.translate_tabs_to_spaces;
+                let tab_size = self.config.items.tab_size;
+                let tab_size = if c % tab_size == 0 { tab_size } else { c % tab_size };
```

Comment:
I think this is reasonable, but another way to write it is `(c + tab_size - 1) % tab_size + 1`. Your way is probably clearer though. I actually looked on godbolt and the code is _slightly_ better for my variant, but I think we're well into "this doesn't really matter" territory.